Userland/Libraries/LibGfx/VectorN.h

```diff
@@ -193,15 +193,19 @@ requires(N >= 2 && N <= 4) class VectorN final {
 
     [[nodiscard]] constexpr T length() const
     {
-        if constexpr (N == 2)
-            return AK::hypot(m_data[0] * m_data[0] + m_data[1] * m_data[1]);
-        else if constexpr (N == 3)
-            return AK::sqrt(m_data[0] * m_data[0] + m_data[1] * m_data[1] + m_data[2] * m_data[2]);
-        else
-            return AK::sqrt(m_data[0] * m_data[0] + m_data[1] * m_data[1] + m_data[2] * m_data[2] + m_data[3] * m_data[3]);
+        T squared_sum { 0 };
+        UNROLL_LOOP
+        for (auto i = 0u; i < N; ++i)
+            squared_sum += m_data[i] * m_data[i];
+        return AK::sqrt(squared_sum);
```

Comment:
This changes it from `hypot` to `sqrt` for `N==2`. I think that is ok, but just want to make sure that is intentional. Please just mark as resolved if that is true.

Reply:
@ldm5180 Well, check out how `AK::hypot` was called: it takes two arguments but only one was provided. Additionally, the argument was squared and `hypot` will square it again. I don't think the original code worked as intended, and this should!

Reply:
The original code should have been `AK::hypot(m_data[0], m_data[1])` but I don't see a good reason to keep it as an exception for `N == 2`.